### ISSUE TITLE
HPACK test case 5.2.3 is wrong

### ISSUE THREAD

**Issue by cesarbs** [summerwind/h2spec#82]
Please correct me if I'm wrong, but I believe [that test](https://github.com/summerwind/h2spec/blob/d99d17171179c709cf23a3df1aadc9657addf825/hpack/5_2_string_literal_representation.go#L113) is not sending an EOS anywhere.

It sends a HEADERS frame with the following header representation in it:

```
"\x00\x85\xf2\xb2\x4a\x87\xff\xff\xff\xfd\x25\x42\x7f"
```

That decodes as:

* 0x00 - Literal Header Field Representation without Indexing - New Name
* 0x85 - Header name has 5 octets and is Huffman encoded
* (0xf2, 0xb2, 0x4a, 0x87, 0xff) decodes as "x-tes9". More specifically:

```
x      -      t     e    s     9      *
11110010 10110010 01001010 10000111 11111111
```

Where `*` marks the beginning of padding.

So nothing wrong in the header name (except that it decodes as "x-tes9" instead of "x-test" as it says in [the comment](https://github.com/summerwind/h2spec/blob/d99d17171179c709cf23a3df1aadc9657addf825/hpack/5_2_string_literal_representation.go#L124).

Moving on to the header value, it is encoded as a Huffman-encoded 622462-octet sequence, which is clearly incomplete.

I think the sequence `"\xff\xff\xff\xfd"` in the input data was meant to encode the EOS symbol, but that's not what's happening in that test.

**Comment by cesarbs** [comment]
This is the test description:

```
Desc:        "Sends a Huffman-encoded string literal representation containing the EOS symbol",
Requirement: "The endpoint MUST treat this as a decoding error.",
```

Looks like it should be sending an (invalid) EOS and checking the server signals an error.

**Comment by nitely** [comment]
Ah, sorry, I see what you mean now. It seems the name length is wrong.

**Comment by summerwind** [comment]
Thank you for the report! 
You are correct. This encoded string is wrong and fixed in ce109b0.
